### PR TITLE
netlayers/tcp: init

### PR DESCRIPTION
**Pull request by cab404**
Support for trivial TCP netlayer, useful for quick tests and experimentation. Addresses look like this: `ocapn://localhost:22222.tcp`.

### CODE REVIEW

**Comment by dpwiz** [comment]
Did you paint it red somewhere? It's goes much faster this way (=

**Comment by cab404** [comment]
@tsyesika ..?

**Comment by tsyesika** [comment]
Hey,

Sorry it's taken me a while to get to see this. Thanks for providing the PR.

I think we need something like this as the onion netlayer while being useful in the real world, isn't the best choice when thinking about test suites as it's kind of slow and sometimes flaky. I wonder if before adding this netlayer to the test suite, if we should propose it (or something like it) to the group and have it documented in the [draft netlayer spec](https://github.com/ocapn/ocapn/blob/main/draft-specifications/Netlayers.md). Could I suggest you create a PR to add documentation and the OCapN group can then weigh in?

I don't know if you've seen the meetings, they're on the 2nd Tuesday of every month, you're welcome to join.

File: netlayers/tcp.py
Comment (cab404) [inline netlayers/tcp.py:36]: add <65536 condition

**Comment by cab404** [comment]
> Hey,
> 
> Sorry it's taken me a while to get to see this. Thanks for providing the PR.
> 
> I think we need something like this as the onion netlayer while being useful in the real world, isn't the best choice when thinking about test suites as it's kind of slow and sometimes flaky. I wonder if before adding this netlayer to the test suite, if we should propose it (or something like it) to the group and have it documented in the [draft netlayer spec](https://github.com/ocapn/ocapn/blob/main/draft-specifications/Netlayers.md). Could I suggest you create a PR to add documentation and the OCapN group can then weigh in?
> 
> I don't know if you've seen the meetings, they're on the 2nd Tuesday of every month, you're welcome to join.

Sure thing, I will rebase and add documentation. Just wanted to make sure it's not getting forgotten about altogether :)

File: netlayers/tcp.py
Comment (dpwiz) [reply netlayers/tcp.py:36]: can't you just bind to port 0 and get one from the system?

Comment (dpwiz) [reply netlayers/tcp.py:36]: yes we can (=

```python
import socket
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(('0.0.0.0', 0))
s.getsockname()[1]
50189
```

**Comment by cab404** [comment]
Hi there! Any chance this getting merged?

**Comment by cab404** [comment]
@tsyesika

File: README.md
Comment (tsyesika) [inline README.md:15]: We should rename this to something like "TCP Testing netlayer (Do not use in production)!"

File: netlayers/tcp.py
Comment (tsyesika) [inline netlayers/tcp.py:48]: We should probably change the transport name to something which makes it clear this is for testing, something like "tcp-testing-only://"

**Review by tsyesika — CHANGES_REQUESTED** [review]
I did want to get the groups opinion on this before merging but we've not had a discussion about it within the meeting yet. That said it's been a while, I suspect we can merge this and if the group want some modifications we can always make those later.

I think this a much better netlayer to use for testing, but since it doesn't have any form of encryption, doesn't protect against MITM attacks, etc. We should probably make it clear this is just for testing. I think it makes sense to do this both in the README and in the transport name like I suggested in my inline comments.

File: README.md
Comment (cab404) [reply README.md:15]: Sure, «testing netlayer» in «ocapn-test-suite» it is)

**Comment by cab404** [comment]
You are free to introduce any changes to make it mergeable~

**Pull request merged**
